### DISCUSSION TITLE
How to use UseSuspenseQueryResult type

### DISCUSSION THREAD

**Discussion by nilshartmann** [TanStack/query#6574] [Q&A] [answered]
Hi all,

I'm wondering, why this code snippet do not raise a typescript error, even `doFetch` returns a string, but `UseSuspenseQueryResult` declares a `number`.

```typescript
  async function doFetch() {
    return "happy holidays ☃️"
  }
  
  function useGetBestWishesQuery(): UseSuspenseQueryResult<number> {
    return useSuspenseQuery({
      queryKey: ["..."],
      queryFn: () => doFetch(),
    });
  }
```

Only if I add a type parameter to `useSuspenseQuery` (`useSuspenseQuery<number>`) I get expected type errrors.

(Query v5.14.2, TypeScript 5.2.2)

Thanks for any hints!

**Comment by TkDodo** [comment, marked as answer]
hm, weird. the same issue for `useQuery`.

https://www.typescriptlang.org/play?#code/ATCWFsAcHsCcBdgG9gFcDOBTAiqzsBPAGmAFUtd8CAlTdVAG0QF9gAzWacYAcgAF4AQwB26IQGMA1gHpYmQePgBaAI55CPAFAhtwQegLDx7VEfihow4ABNoAMUzxxACwAUASmS6Qc+KlhWAETOgpCQBMDO0Ayg1oIE6MCAwGSA8H+Busy6umymihZWGJgA4o4AQnTwAOqg6M50lIQeAFxkFOo0dIzwADzCqOAARvgAfF4gPo7+BW1Urkje42pUANKYBC0A2oEAdLuBALpECyBLhHbCLR7AALyjtg5Obu5H48DM7gDcGZq64pZiyBsgiEbxuaCwJXg5TE1Vq9XaHl00mkrxAAD0APxAA

It actually also behaves the same way back to v3

Maybe file an issue but my recommendation is to use type inference and not annotate the result of `useQuery`

**Reply by nilshartmann** [reply]
Thanks for your quick reply! I think fixing this somehow would make probably sense, because otherwise you get false safety in case you're specifing the return type explizitly.

**Reply by TkDodo** [reply]
Please feel free to look into it if it's important to you, I don't know off the top of my head what the issue could be
